### PR TITLE
Create Feature Stages page in Main, 2.25 and 2.26

### PR DESCRIPTION
**Pull request by csengerszabo**
Adding a Feature Stages page to the Kubermatic Documentation.

### CODE REVIEW

**Review by xmudrii — APPROVED** [review]
/lgtm
/approve
/hold
for other folks to review (cc @xrstf @embik)

**Comment by xrstf** [comment]
So this does not apply to KKP 2.26?

**Comment by csengerszabo** [comment]
> So this does not apply to KKP 2.26?

@xrstf good point, I added it now.

File: content/kubermatic/main/architecture/feature-stages/_index.en.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+- The schema and/or semantics of objects may change in incompatible ways in later software release
+  - When this happens, we will provide instructions for migrating to the next version, which can require manual work and/or downtime
+- The whole feature can still be revoked, but with prior notice and respecting a deprecation cycle
+- Recommended for only non-business-critical uses, testing usability, performance, and compatibility in real-world environments
+
+
+## General Availability (GA)
+
+- Users: All users
+- The feature is always enabled; you cannot disable it by a feature gate
```
Comment (xrstf) [inline content/kubermatic/main/architecture/feature-stages/_index.en.md:34]: KKP does have some feature gates, like HeadlessInstallation / UserClusterMLA / DevelopmentEnvironment / VerticalPodAutoscaler, which are GA and not enabled by default. This document doesn't yet consider optional features which are meant to be optional.

Comment (csengerszabo) [reply content/kubermatic/main/architecture/feature-stages/_index.en.md:34]: Do all the other GA traits apply to these Optional Features? If yes, then I can just add a sub-item here noting that Optional Features are an exception of this. CC @xmudrii

Comment (xmudrii) [reply content/kubermatic/main/architecture/feature-stages/_index.en.md:34]: Ideally speaking, this shouldn't happen, there should be other ways to handle those features. But that's ideally, and making this work would require API changes. Given that, let's add a sub-item noting about optional features.

Comment (csengerszabo) [reply content/kubermatic/main/architecture/feature-stages/_index.en.md:34]: Added the line

**Comment by csengerszabo** [comment]
@xmudrii @xrstf can you take a second look and approve if all good?

**Review by embik — APPROVED** [review]
@xmudrii @xrstf please lift the hold when you're okay with this PR.

**Pull request merged**